keras_nlp/src/models/mobilenet/mobilenet_backbone.py

```diff
@@ -0,0 +1,517 @@
+                are used at each layer.
+        input_filter: number of filters in first convolution layer
+        output_filter: specifies whether to add conv and batch_norm in the end,
+            if set to None, it will not add these layers in the end.
+            'None' for MobileNetV1
+        activation: activation function to be used
+            'hard_swish' for MobileNetV3,
+            'relu6' for MobileNetV1 and MobileNetV2
+        inverted_res_block: whether to use inverted residual blocks or not,
+            'False' for MobileNetV1,
+            'True' for MobileNetV2 and MobileNetV3
+
+
+    Example:
+    ```python
+    input_data = tf.ones(shape=(8, 224, 224, 3))
+
+    # Randomly initialized backbone with a custom config
+    model = MobileNetBackbone(
+        stackwise_expansion=[1, 4, 6],
+        stackwise_filters=[4, 8, 16],
+        stackwise_kernel_size=[3, 3, 5],
+        stackwise_stride=[2, 2, 1],
+        stackwise_se_ratio=[0.25, None, 0.25],
+        stackwise_activation=["relu", "relu6", "hard_swish"],
+        include_rescaling=False,
+        output_filter=1280,
+        activation=keras.activations.hard_swish,
+        inverted_res_block=True,
+
+    )
+    output = model(input_data)
+    ```
+    """
+
+    def __init__(
+        self,
+        stackwise_expansion,
+        stackwise_filters,
+        stackwise_kernel_size,
+        stackwise_stride,
+        stackwise_se_ratio,
+        stackwise_activation,
+        include_rescaling,
+        output_filter,
+        inverted_res_block,
+        image_shape=(224, 224, 3),
+        activation=keras.activations.hard_swish,
+        depth_multiplier=1.0,
+        input_filters=16,
+        **kwargs,
+    ):
+        # === Functional Model ===
+        CHANNEL_AXIS = (
+            -1 if keras.config.image_data_format() == "channels_last" else 1
+        )
+
+        inputs = keras.layers.Input(shape=image_shape)
+        x = inputs
+
+        if include_rescaling:
+            x = keras.layers.Rescaling(scale=1 / 255)(x)
+
+        input_filters = adjust_channels(input_filters)
+        x = keras.layers.Conv2D(
+            input_filters,
+            kernel_size=3,
+            strides=(2, 2),
+            padding="same",
+            data_format=keras.config.image_data_format(),
+            use_bias=False,
+            name="input_conv",
+        )(x)
+        x = keras.layers.BatchNormalization(
+            axis=CHANNEL_AXIS,
+            epsilon=BN_EPSILON,
+            momentum=BN_MOMENTUM,
+            name="input_batch_norm",
+        )(x)
+        x = keras.layers.Activation(
+            activation,
+        )(x)
+
+        for stack_index in range(len(stackwise_filters)):
+
+            if inverted_res_block:
+                x = apply_inverted_res_block(
+                    x,
+                    expansion=stackwise_expansion[stack_index],
+                    filters=adjust_channels(
```

Comment:
consider pull this out of the if block, it's the same in both cases.
`filters = adjust_channels(stackwise_filters[stack_index] * depth_multiplier)`